pkg/protocol/protocol_fork.go

```diff
@@ -213,9 +212,6 @@ func (p *Protocol) processFork(fork *chainmanager.Fork) (anchorBlockIDs iotago.B
 	defer unhook()
 	defer wp.Shutdown()
 
-	ctx, cancel := context.WithTimeout(p.context, 2*time.Minute)
-	defer cancel()
```

Comment:
Taking the parent context `p.context` and getting a new one with a timeout is actually necessary here as the child context gets also cancelled if the parent gets cancelled. This is important for shutting down the node properly without hanging this goroutine until the final timeout passed. Also this whole timeout is for limiting the time we will try to process the fork. There might be hundreds of slots to verify etc so that sets an upper limit to that as well